app/src/views/AnalysisSetupView.js

```diff
@@ -119,7 +119,10 @@
         },
 
         render: function () {
-            this.inputsView.collection.set(this.model.get('meta').analysis.inputs);
+            if (this.inputsView.analysisName !== this.model.get('meta').analysis.name) {
+                this.inputsView.analysisName = this.model.get('meta').analysis.name;
+                this.inputsView.collection.set(this.model.get('meta').analysis.inputs);
+            }
```

Comment:
This check is not strong enough. If you edit the inputs (e.g. add an input) and click Save, then click Setup and run, the view no longer updates with the new input field. Any change to the inputs should trigger a full reset - no need to remember any inputs that happen to match.

I think we need something like this:
```js
// Deep comparison
if (!_.isEqual(this.inputsView.inputs, this.model.get('meta').analysis.inputs)) {
    // Deep copy
    this.inputsView.inputs = jQuery.extend({}, this.model.get('meta').analysis.inputs);
    this.inputsView.collection.set(this.inputsView.inputs);
}
```